p2pc/main.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import "C"
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+
+	peer "github.com/libp2p/go-libp2p-peer"
+	identify "github.com/libp2p/go-libp2p/p2p/protocol/identify"
+	multiaddr "github.com/multiformats/go-multiaddr"
+
+	"github.com/libp2p/go-libp2p-daemon/p2pclient/go"
+)
+
+// ClientConfig defines the configuration options
+type ClientConfig struct {
+	pathd   *string
+	pathc   *string
+	command *string
+	args    []string
+}
+
+func main() {
+	identify.ClientVersion = "p2pc/0.1"
+	config := initialize()
+	start(config)
+}
+
+func initialize() ClientConfig {
+	config := ClientConfig{
+		pathd:   flag.String("pathd", "/tmp/p2pd.sock", "daemon control socket path"),
+		pathc:   flag.String("pathc", "/tmp/p2pc.sock", "client control socket path"),
+		command: flag.String("command", "Identify", "command to send to the daemon"),
+	}
+	flag.Parse()
+	config.args = flag.Args()
+	// delete control socket if it already exists
+	if _, err := os.Stat(*config.pathc); !os.IsNotExist(err) {
+		err = os.Remove(*config.pathc)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+	return config
+}
+
+func start(config ClientConfig) {
+	client, err := p2pclient.NewClient(*config.pathd, *config.pathc)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if *config.command == "Identify" {
```

Comment:
Prefer to use a switch over the string.

Reply:
i have several other fixes and cleanup to add.  do you mind if i open another PR ore should I fix these here?

Reply:
Better if you can fix it here :-)

Reply:
done